cpp/src/arrow/flight/sql/odbc/odbc_api.cc

```diff
@@ -144,6 +149,12 @@ SQLRETURN SQLGetDiagFieldW(SQLSMALLINT handleType, SQLHANDLE handle,
   using ODBC::ODBCConnection;
   using ODBC::ODBCEnvironment;
 
+  LOG_DEBUG(
```

Comment:
We have a lot of logs for `SQLGetDiagFieldW` and `SQLGetDiagRecW` due to an issue with incorrectly reported truncation which is fixed in #38